File: src/main/org/nlogo/prim/_letvariable.java
package org.nlogo.prim ;

import org.nlogo.nvm.Context;
import org.nlogo.api.Let;
import org.nlogo.nvm.Reporter;
import org.nlogo.nvm.Syntax;

public final strictfp class _letvariable
	extends Reporter
{
	public final Let let ;
	final String name ;
	public _letvariable( Let let , String name )
	{
		this.let = let ;
		this.name = name ;
	}
	@Override public Syntax syntax()
	{
		return Syntax.reporterSyntax( Syntax.TYPE_WILDCARD ) ;
	}
	@Override public String toString()
	{
		return super.toString() + "(" + name + ")";
	}
	@Override public Object report( Context context )
	{
		return report_1( context ) ;
	}
	public Object report_1( Context context )
	{
		return context.getLet( let ) ;
	}
}

File: headless/src/main/org/nlogo/prim/_letvariable.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim

import org.nlogo.api.{ Syntax, Let }
import org.nlogo.nvm.{ Context, Reporter }

class _letvariable(_let: Let) extends Reporter {

  // MethodRipper won't let us call a public method from perform_1() - ST 7/20/12
  def let = _let

  override def syntax =
    Syntax.reporterSyntax(Syntax.WildcardType)
  override def toString =
    super.toString + "(" + _let.name + ")"

  override def report(context: Context) =
    report_1(context)
  def report_1(context: Context) =
    context.getLet(_let)

}
